libcxx/include/__algorithm/generate.h

```diff
@@ -20,8 +21,7 @@ _LIBCPP_BEGIN_NAMESPACE_STD
 template <class _ForwardIterator, class _Generator>
 inline _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX20 void
 generate(_ForwardIterator __first, _ForwardIterator __last, _Generator __gen) {
-  for (; __first != __last; ++__first)
-    *__first = __gen();
+  std::for_each(__first, __last, [&](auto& __element) { __element = __gen(); });
```

Comment:
We need to handle iterators dereferencing to rvalues, as indicated by the error messages.
```suggestion
  std::for_each(__first, __last, [&](auto&& __element) { static_cast<decltype(__element)>(__element) = __gen(); });
```

Reply:
> Just to clarify, is it because when dereferencing `LifetimeIterator`, we get a type named `Reference` which is itself an rvalue, so `auto&` cannot bind to it?

Yes, it is.